test/server/cards/09.5-aCF/UnifiedCompany.spec.js

```diff
@@ -0,0 +1,71 @@
+describe('Unified Company', function() {
+    integration(function() {
+        describe('Unified Company\'s ability', function() {
+            beforeEach(function() {
+                this.setupTest({
+                    phase: 'conflict',
+                    player1: {
+                        inPlay: ['unified-company'],
+                        hand: ['fine-katana'],
+                        dynastyDiscard: ['solemn-scholar', 'akodo-gunso', 'agasha-taiko', 'borderlands-defender']
+                    },
+                    player2: {
+                        hand: ['ornate-fan', 'banzai'],
+                        dynastyDiscard: ['brash-samurai']
+                    }
+                });
+
+                this.unifiedCompany = this.player1.findCardByName('unified-company');
+                this.scholar = this.player1.findCardByName('solemn-scholar');
+                this.gunso = this.player1.findCardByName('akodo-gunso');
+                this.taiko = this.player1.findCardByName('agasha-taiko');
+                this.borderlands = this.player1.findCardByName('borderlands-defender');
+
+                this.brash = this.player2.findCardByName('brash-samurai');
+
+                this.noMoreActions();
+                this.initiateConflict({
+                    type: 'military',
+                    attackers: [this.unifiedCompany],
+                    defenders: []
+                });
+            });
+
+            it('should trigger once the character wins a conflict and the controller has less cards in hand', function() {
+                this.noMoreActions();
+                expect(this.player1).toHavePrompt('Triggered Abilities');
+                this.player1.clickCard(this.unifiedCompany);
+                expect(this.player1).toHavePrompt('Unified Company');
+            });
+
+            it('should only be able to target non-unique cost 2 or less bushi in your dynasty discard pile', function() {
+                this.noMoreActions();
+                this.player1.clickCard(this.unifiedCompany);
+                expect(this.player1).toBeAbleToSelect(this.gunso);
+                expect(this.player1).not.toBeAbleToSelect(this.taiko);
+                expect(this.player1).not.toBeAbleToSelect(this.borderlands);
+                expect(this.player1).not.toBeAbleToSelect(this.brash);
+                expect(this.player1).not.toBeAbleToSelect(this.scholar);
+            });
+
+            it('should put the character into play', function() {
+                expect(this.gunso.location).toBe('dynasty discard pile');
+                this.noMoreActions();
+                this.player1.clickCard(this.unifiedCompany);
+                this.player1.clickCard(this.gunso);
+                expect(this.gunso.location).toBe('play area');
+                expect(this.gunso.isParticipating()).toBe(false);
+            });
+
+            it('should not trigger if you have equal or less cards in hand', function() {
```

Comment:
I would split this test case into "should not trigger if you have equal cards in hand" and "should not trigger if you have more cards in hand".